tests/unittests/test_shared_memory_map.py

```diff
@@ -0,0 +1,131 @@
+            mem_map_name = self.get_new_mem_map_name()
+            mem_map_size = content_size + consts.CONTENT_HEADER_TOTAL_BYTES
+            mem_map = self.file_accessor.create_mem_map(mem_map_name,
+                                                        mem_map_size)
+            shared_mem_map = SharedMemoryMap(self.file_accessor, mem_map_name,
+                                             mem_map)
+            content = self.get_random_bytes(content_size)
+            num_bytes_written = shared_mem_map.put_bytes(content)
+            self.assertEqual(content_size, num_bytes_written)
+            dispose_status = shared_mem_map.dispose()
+            self.assertTrue(dispose_status)
+
+    def test_get_bytes(self):
+        """
+        Create a SharedMemoryMap, write bytes to it and then read them back.
+        Verify that the bytes written and read match.
+        """
+        for content_size in [1, 10, 1024, 2 * 1024 * 1024, 20 * 1024 * 1024]:
+            mem_map_name = self.get_new_mem_map_name()
+            mem_map_size = content_size + consts.CONTENT_HEADER_TOTAL_BYTES
+            mem_map = self.file_accessor.create_mem_map(mem_map_name,
+                                                        mem_map_size)
+            shared_mem_map = SharedMemoryMap(self.file_accessor, mem_map_name,
+                                             mem_map)
+            content = self.get_random_bytes(content_size)
+            num_bytes_written = shared_mem_map.put_bytes(content)
+            self.assertEqual(content_size, num_bytes_written)
+            read_content = shared_mem_map.get_bytes()
+            self.assertEqual(content, read_content)
+            dispose_status = shared_mem_map.dispose()
+            self.assertTrue(dispose_status)
+
+    def test_put_bytes_more_than_capacity(self):
+        """
+        Attempt to put more bytes into the created SharedMemoryMap than the
+        size with which it was created. Verify that an SharedMemoryException is
+        raised.
+        """
+        mem_map_name = self.get_new_mem_map_name()
+        mem_map_size = 1024 + consts.CONTENT_HEADER_TOTAL_BYTES
+        mem_map = self.file_accessor.create_mem_map(mem_map_name,
+                                                    mem_map_size)
+        shared_mem_map = SharedMemoryMap(self.file_accessor, mem_map_name,
+                                         mem_map)
+        # Attempt to write more bytes than the size of the memory map we created
+        # earlier (1024).
+        content_size = 2048
+        content = self.get_random_bytes(content_size)
+        with self.assertRaisesRegex(ValueError, 'out of range'):
+            shared_mem_map.put_bytes(content)
+        dispose_status = shared_mem_map.dispose()
+        self.assertTrue(dispose_status)
+
+    @unittest.skipIf(os.name == 'nt',
+                     'Windows will create an mmap if one does not exist')
+    def test_dispose_without_delete_file(self):
```

Comment:
```python
―――――――――――――――――――――――――― TestSharedMemoryMap.test_dispose_without_delete_file ―――――――――――――――――――――――――――
Traceback (most recent call last):
  File "/Users/varadmeru/work/microsoft/azfunctions/pyworker/tests/unittests/test_shared_memory_map.py", line 120, in test_dispose_without_delete_file
    shared_mem_map = SharedMemoryMap(self.file_accessor, mem_map_name,
  File "/Users/varadmeru/work/microsoft/azfunctions/pyworker/azure_functions_worker/bindings/shared_memory_data_transfer/shared_memory_map.py", line 25, in __init__
    raise SharedMemoryException(
azure_functions_worker.bindings.shared_memory_data_transfer.shared_memory_exception.SharedMemoryException: Cannot initialize SharedMemoryMap. Invalid memory map provided
```

Reply:
Just assertRaises in this case if this case cannot work for mac.

Reply:
It should, I think. Let's try on your machine and see what is breaking.